plugins/chromatic_correction.py

```diff
@@ -0,0 +1,392 @@
+
+    return scale_cor, translation_cor, rotation_cor, shear_cor
+
+
+def get_file_location(filename, ext):
+    """Get the file location by adding the directory and date time"""
+    dir_path = os.path.join(get_home_folder(), "Documents/Chromatic aberration correction")
+    os.makedirs(dir_path, exist_ok=True)
+    current_date = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    file_location = os.path.join(dir_path, f"{current_date}_{filename}.{ext}")
+    return file_location
+
+
+def update_im_metadata(im: DataArray, scale: [float, float], translation: [float, float], rotation: float,
+                       shear: float):
+    """
+    Update the metadata of the input datarray
+    :param im: image with metadata
+    :param scale: scale correction
+    :param translation: translation correction
+    :param rotation: rotation correction
+    :param shear: shear correction
+    """
+    im.metadata.update({model.MD_PIXEL_SIZE_COR: scale})
+    im.metadata.update({model.MD_POS_COR: translation})
+    im.metadata.update({model.MD_ROTATION_COR: rotation})
+    im.metadata.update({model.MD_SHEAR_COR: shear})
+
+
+def convert_transformation_to_dict(scale: [float, float], translation: [float, float], rotation: float,
+                                   shear: float) -> Dict:
+    """
+    Convert transformation parameters to dictionary
+    :param scale: scale correction
+    :param translation: translation correction
+    :param rotation: rotation correction
+    :param shear: shear correction
+    :return: (Dict) transformation values of the given data
+    """
+    # To enable chromatic correction, add MD_CHROMATIC_COR
+    # under the filter wheel component of the microscope file
+    correction_dict = {
+        model.MD_PIXEL_SIZE_COR: [float(scale[0]), float(scale[1])],
+        model.MD_POS_COR: [float(translation[0]), float(translation[1])],
+        model.MD_ROTATION_COR: float(rotation),
+        model.MD_SHEAR_COR: float(shear)
+    }
+
+    return correction_dict
+
+
+def get_chromatic_correction_dict(im: DataArray, points_ref: numpy.ndarray, points_im: numpy.ndarray):
+    """
+    Get the chromatic correction dictionary for the given data array.
+    :param im: static image values including the metadata
+    :param points_ref: (x, y) location of features in reference data
+    :param points_im: (x, y) location of features in given data by im
+    :return: (Dict) transformation values of the given data
+    """
+    affine = AffineTransform.from_pointset(points_ref, points_im)
+    # Estimate the transformation matrix
+    scale, translation, rotation, shear = extract_transformation_parameters(affine)
+    # Convert translation from pixels to m/physical units
+    pixel_size = im.metadata[model.MD_PIXEL_SIZE]
+    translation = [translation[0] * pixel_size[0], translation[1] * pixel_size[1]]
+    # Correct the metadata of im
+    update_im_metadata(im, scale, translation, rotation, shear)
```

Comment:
It's a little unexpected that the function updates the metadata of the image. Typically, you don't want to change the input parameters. Either indicate it clearly in the docstring (that `im` is being updated), or let the caller code do it (it's just 1 line). I'd favour the second option, but up to you.